app/controllers/stories.controller.js

```diff
@@ -129,39 +130,168 @@ exports.extend = async (req, res) => {
 };
 
 // Retrieve all stories from the database.
-exports.findAll = (req, res) => {
-  const storyId = req.query.storyId;
+exports.findAllForUser = (req, res) => {
+  const userId = req.params.userId;
+  var condition = userId
+    ? {
+      userId: {
+        [Op.like]: `%${userId}%`,
+      },
+    }
+    : null;
+
+  Story.findAll({ where: condition })
+    .then((data) => {
+      res.send(data);
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message:
+          err.message || "Some error occurred while retrieving stories.",
+      });
+    });
+};
+
+exports.findAllParentStoriesForUser = (req, res) => {
+  Story.findAll({
+    where: { userId: req.params.userId, parentId: null },
+    order: [["createdAt", "ASC"]]
+  })
+    .then(data => {
+      res.send(data);
+    })
+    .catch(err => {
+      res.status(500).send({
+        message: err.message || "An error occurred while retrieving stories."
+      });
+    });
+};
+
+// Retrieve all stories from the database for the user
+exports.findOneForUser = (req, res) => {
+  const storyId = req.params.storyId;
+  const userId = req.params.userId;
   var condition = storyId
     ? {
-      id: {
+      storyId: {
         [Op.like]: `%${storyId}%`,
       },
     }
     : null;
 
-  Story.findAll({ where: condition, order: [["name", "ASC"]] })
+  var condition2 = userId
+    ? {
+      userId: {
+        [Op.like]: `%${userId}%`,
+      },
+    }
+    : null;
+
+  Story.findAll({ where: condition, condition2 })
```

Comment:
does this work?  syntax seems awfully suspect

Reply:
Got it fixed up. it wasn't working and I was just getting lucky with the few tests I was making. changed the code and it is working now